WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,181 @@
-
-  <%= link_to "New competition", new_competition_path %>
-  <table class="table">
-    <% @competitions.each do |competition| %>
-      <tr>
-        <td><%= competition.id %></td>
-        <td><%= competition.name %></td>
-        <td><%= link_to "Edit", admin_edit_competition_path(competition) %></td>
-        <td><%= link_to "Results", "/results/competition.php?competitionId=#{competition.id}" %></td>
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <div id="simple-nav" class="col-md-12">
+    <% if params[:display] == "List" %>
+      <ul id="lists-navigation">
+
+        <li class="same-line"><a href="#past" class="js-scroll-to <%= @past_competitions.length > 0 ? '' : 'disabled' %>"><span class="glyphicon glyphicon-chevron-left" aria-hidden="true" data-toggle="tooltip" data-placement="bottom" data-container="body" title="<%= @past_competitions.length > 0 ? 'Past competitions' : 'No past competitions selected' %>"></span></a></li>
+
+        <li class="same-line"><a href="#current" class="js-scroll-to <%= @in_progress_competitions.length > 0 ? '' : 'disabled' %>"><span class="glyphicon glyphicon-time" aria-hidden="true" data-toggle="tooltip" data-placement="bottom" data-container="body" title="<%= @in_progress_competitions.length > 0 ? 'Competitions in progress' : 'No competitions in progress' %>"></span></a></li>
+
+        <li class="same-line"><a href="#upcoming" class="js-scroll-to <%= @upcoming_competitions.length > 0 ? '' : 'disabled' %>"><span class="glyphicon glyphicon-chevron-right" aria-hidden="true" data-toggle="tooltip" data-placement="bottom" data-container="body" title="<%= @upcoming_competitions.length > 0 ? 'Upcoming competitions' : 'No upcoming competitions selected' %>"></span></a></li>
+
+        <li class="same-line last-one"><a href="#comp-query-form" class="js-scroll-to">Top</a></li>
+      </ul>
     <% end %>
-  </table>
+  </div>
+
+  <%= form_tag competitions_path, method: :get, class: "competition-select form-inline", id: "comp-query-form" do %>
+    <%= render 'index_form' %>
+  <% end %>
+
+  <% if @competitions.length == 0 %>
+    <div class="alert alert-warning">No competitions found.</div>
+  <% else %>
+    <% if params[:display] == "List" %>
+      <div class="row competition-list">          
+        <% if @in_progress_competitions.length > 0 %>
+          <div class="col-md-12" id="current">  
+            <%= render 'index_table', competitions: @in_progress_competitions, title: "Competitions in progress" %>
+          </div>
+        <% end %>
+
+        <% if @upcoming_competitions.length > 0 %>
+          <div class="col-md-12" id="upcoming">
+            <%= render 'index_table', competitions: @upcoming_competitions, title: "Upcoming competitions" %>
+          </div>
+        <% end %>
+
+        <% if @past_competitions.length > 0 %>
+          <div class="col-md-12" id="past">
+            <%= render 'index_table', competitions: @past_competitions, title: "Past competitions (last 90 days)" %>
```

Comment:
Let's use `CompetitionsController::PAST_COMPETITIONS_DAYS` here instead of copying the number 90 around.